jest/private/bazel_haste_map.cjs

```diff
@@ -0,0 +1,106 @@
+const { join, dirname, extname } = require("path");
+const { readFileSync } = require("fs");
+
+// to require jest-haste-map we need to hop from jest-cli => @jest/core => jest-haste-map in the virtual store
+const jestCliPackage = dirname(require.resolve("jest-cli/package.json"));
+const jestConfigPackage = dirname(
+  require.resolve(join(jestCliPackage, "../@jest/core/package.json")),
+);
+const HasteMap = require(
+  join(jestConfigPackage, "../../jest-haste-map"),
```

Comment:
Consider having the user depend on jest haste map and passing it in as a label instead. Not sure that's better, but we should at least consider if it's better

Reply:
We essentially don't want the user to even know of haste-map really, we're almost working around the fact that jest even uses it.

The other `bazel_*.cjs` files in this directory use the same mechanism to access these packages so this isn't new here.